source/set-return-type.d.ts

```diff
@@ -0,0 +1,29 @@
+type IsAny<T> = 0 extends (1 & T) ? true : false; // https://stackoverflow.com/a/49928360/3406963
+type IsNever<T> = [T] extends [never] ? true : false;
+type IsUnknown<T> = IsNever<T> extends false ? T extends unknown ? unknown extends T ? IsAny<T> extends false ? true : false : false : false : false;
```

Comment:
Do you think it's worth exposing any of these types? (Could be done later on if decided on. Just curious about your opinion).

Reply:
I would really like that, personally! Whenever I want to use any of them (mostly for testing purposes), I have to look them up around the internet, they're not so easy to memorize. If they were here I'd just quickly import instead :grimacing: I use them when trying to build some complex type every now and then. I think I used `IsAny` somewhere else in a PR to another repository of yours...

It may be a little hard to give examples of real use cases though... It might be fun to try to find good examples... Maybe just saying that they can be used to work around edge cases when creating more complex types?